Fix pretty printing of symbols by Test.jl

nerdberg792 opened the pull request:
Fixes #57287
When we perform a Test : 
```
julia> x = 1; y = :sym; @test x == y
Test Failed at REPL[48]:1
  Expression: x == y
   Evaluated: 1 == sym
```
But ideally the test should print `Evaluated 1 == :sym`

I have changed the parsing of the arguments :
---

Updated Test.jl and added some if else conditions : 

now the tests mentioned in the issue yields these results : 

```
julia> x = 1; y = :sym; @test x == y
Test Failed at REPL[2]:1
  Expression: x == y
   Evaluated: 1 == :sym

```

```
julia> x = 1; y = :(1+1); @test x == y
Test Failed at REPL[2]:1
  Expression: x == y
   Evaluated: 1 == :(1 + 1)

```

nerdberg792 committed:
updated Test.jl

DilumAluthge commented:
Can you edit the PR title to be a bit more descriptive, e.g. to briefly describe the bug that is being fixed.

nerdberg792 committed:
Update Test.jl

Removed whitespaces

nerdberg792 commented:
> Can you edit the PR title to be a bit more descriptive, e.g. to briefly describe the bug that is being fixed.

I have made some changes can you review my PR , also I haven't added any tests to check for the aforementioned bug

nerdberg792 committed:
Merge branch 'master' into pretty

nerdberg792 committed:
Update Test.jl

removed whitespaces

nerdberg792 committed:
Update Test.jl

removed whitespace from line 367

nerdberg792 committed:
Update Test.jl

removeed whitespaces

nerdberg792 commented:
@KristofferC can you please point out where should I add the tests

LilithHafner commented:
stdlib/Test/test/runtests.jl

KristofferC commented on stdlib/Test/src/Test.jl:
```diff
@@ -361,8 +361,20 @@ function eval_test(evaluated::Expr, quoted::Expr, source::LineNumberNode, negate
-            quoted_args[i+2] = b
+            if a isa Symbol
+                quoted_args[i] = QuoteNode(a)
+            elseif Meta.isexpr(a, :call)
```
In which case is this used? What difference does it make?

vtjnash commented on stdlib/Test/src/Test.jl:
```diff
@@ -361,8 +361,20 @@ function eval_test(evaluated::Expr, quoted::Expr, source::LineNumberNode, negate
-            quoted_args[i] = a
-            quoted_args[i+2] = b
+            if a isa Symbol
+                quoted_args[i] = QuoteNode(a)
```
That function is the existing `Base.quoted` function in this case

nerdberg792 committed:
Update Test.jl

On the previous changes , I misinterpret how the expressions need to be printed , fixed this here

nerdberg792 commented:
> The last example in the first post is is already what is shown on master so not sure what is meant by that.

I misinterpreted your comments on the issues , I have now made the necessary changes and it seems to work fine , can you please review this once , sorry for the inconvenience cause !

LilithHafner committed:
Merge branch 'master' into pretty

LilithHafner commented:
This looks good so far. Would you please address Jameson's [comment](https://github.com/JuliaLang/julia/pull/57430#discussion_r1958536633) by using `Base.quoted` and Kristoffer's comment by adding tests that fail on master and pass on this PR?
